### PR TITLE
Add pull request template

### LINKED ISSUE #4002: Create a PR template to help improve ease of reviewing
As @kirawi described in [this comment](https://github.com/helix-editor/helix/discussions/3815#discussioncomment-3672484), making a PR template to have contributors explain the functionality and motivation of their contribution could help make reviewing PRs easier/faster, which is something that seems to be becoming increasingly necessary with the rate at which PRs being opened.

### PR DESCRIPTION
**Pull request by poliorcetics**
Adding a Pull Request template will make it easier for new and repeat contributors to learn and
remember common tasks to do when creating a PR and for reviewers to know what the PR is about with
a quick look.

Closes #4002

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- [ ] Language changed/added (highlights, queries, injections, config in `languages.toml` ...)
+- [ ] Refactor
+- [ ] Theme changed/added
+- [ ] UI change
+
+-----
+
+## Tasks (not applicable to all PRs)
+
+If something is unneeded, just don't check it, if it was needed, either the CI or a reviewer will
```
Comment (SoraTenshi) [inline .github/pull_request_template.md:27]:
```suggestion
If something is not required, just don't check it, if it was required, either the CI or a reviewer will
```
i believe that required sounds a bit better in this context :)

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+<!--- Especially needed if you introduce a new feature or language --->
+- [ ] *(Doc)* I ran `cargo xtask docgen` to ensure the generated documentation is up-to-date
+- [ ] *(Doc)* I updated the user-facing documentation under `book/`
+- [ ] *(Languages)* I ran `cargo xtask query-check` <!--- For when you add/modify a new language --->
+- [ ] *(Themes)* I ran `cargo xtask themelint` <!--- For when you add/modify a new theme --->
+- [ ] *(User-facing changes)* I added an entry to the [`CHANGELOG.md`](https://github.com/helix-editor/helix/blob/master/CHANGELOG.md)
+
+## Questions
+
+<!--- Do you have points you're not sure about ? Add them here --->
```
Comment (SoraTenshi) [inline .github/pull_request_template.md:39]:
```suggestion
<!--- Do you have points you're not sure about? Add them here --->
```

**Review by SoraTenshi — COMMENTED** [review]
My 2 cents :)

**Comment by the-mikedavis** [comment]
I was in favor of a template but I've used them at work for a while now and now I'm not sure I really like them: they add a lot of boilerplate that obscures the actual description.

I think it might be ok though to have a PR template that only includes an HTML comment that gives some pointers like `cargo xtask docgen`. But the github UI brings up the [`CONTRIBUTING`](https://github.com/helix-editor/helix/blob/master/docs/CONTRIBUTING.md) doc for new contributions so it might be better just to clean up and expand that documentation

**Pull request closed without merging**